### ISSUE TITLE
Additional options specified with an object causing rule errors

### ISSUE THREAD

**Issue by ArcticZeroo** [eslint/eslint#10437]
**Tell us about your environment**

* **ESLint Version:**
4.18.2

* **Node Version:**
8.11.2

* **npm Version:**
5.6.0

**What parser (default, Babel-ESLint, etc.) are you using?**
As far as I know, default. I'm using the ESLint built into JetBrains IDEs, but since it uses the eslint executable I would imagine they're identical.

**Please show your full configuration:**
You can also find it in my repo here: https://github.com/ArcticZeroo/carabiner/blob/cb445cdc30ba7791faa335100da8d2b7a11f1131/.eslintrc.json

<details>
<summary>Configuration</summary>

```js
{
  "env": {
    "es6": true,
    "node": true,
    "mocha": true
  },
  "extends": "eslint:recommended",
  "rules": {
    "indent": [
      "error", 4, { "SwitchCase": 1 }
    ],
    "linebreak-style": [
      "error",
      "windows"
    ],
    "quotes": [
      "error",
      "single"
    ],
    "semi": [
      "error",
      "always"
    ],
    "object-curly-spacing": [
      "error",
      "always"
    ],
    "arrow-spacing": [
      "error",
      {
        "before": true,
        "after": true
      }
    ],
    "accessor-pairs": "error",
    "array-callback-return": "error",
    "no-else-return": "error",
    "no-empty-function": "warn",
    "no-implicit-coercion": [
      "error",
      {
        "allow": ["!!"]
      }
    ],
    "no-return-await": "error",
    "no-self-assign": "error",
    "no-throw-literal": "error",
    "require-await": "warn",
    "yoda": [
      "error",
      "never",
      {
        "exceptRange": true
      }
    ],
    "no-shadow": [
      "error"
    ],
    "global-require": "warn",
    "handle-callback-err": [
      "error"
    ],
    "camelcase": [
      "error"
    ],
    "one-var": [
      "error",
      {
        "initialized": "never",
        "uninitialized": "consecutive"
      }
    ],
    "require-jsdoc": [
      "warn",
      {
        "FunctionDeclaration": true,
        "MethodDefinition": true
      }
    ],
    "generator-star-spacing": [
      "error",
      {
        "before": true,
        "after": false
      }
    ],
    "no-var": [
      "error"
    ],
    "object-shorthand": [
      "error",
      "always",
      {
        "avoidQuotes": true

      }
    ],
    "prefer-rest-params": [
      "error"
    ],
    "prefer-template": [
      "error"
    ]
  },
  "parserOptions": {
    "ecmaVersion": 2017
  }
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**
All of the source code. ESLint can't even parse the config when WebStorm attempts to run a full code inspection.

**What did you expect to happen?**
ESLint parses the config.

**What actually happened? Please include the actual, raw output from ESLint.**
```
Error: H:\projects\node\carabiner\.eslintrc.json:
	Configuration for rule "one-var" is invalid:
	Value {"initialized":"never","uninitialized":"consecutive"} should be equal to one of the allowed values.
	Value {"initialized":"never","uninitialized":"consecutive"} should NOT have additional properties.
	Value "consecutive" should be equal to one of the allowed values.
	Value {"initialized":"never","uninitialized":"consecutive"} should match exactly one schema in oneOf.


Error: H:\projects\node\carabiner\.eslintrc.json:
	Configuration for rule "one-var" is invalid:
	Value {"initialized":"never","uninitialized":"consecutive"} should be equal to one of the allowed values.
	Value {"initialized":"never","uninitialized":"consecutive"} should NOT have additional properties.
	Value "consecutive" should be equal to one of the allowed values.
	Value {"initialized":"never","uninitialized":"consecutive"} should match exactly one schema in oneOf.

    at validateRuleOptions (H:\projects\node\carabiner\node_modules\eslint\lib\config\config-validator.js:119:19)
    at Object.keys.forEach.id (H:\projects\node\carabiner\node_modules\eslint\lib\config\config-validator.js:162:9)
    at Array.forEach (<anonymous>)
    at validateRules (H:\projects\node\carabiner\node_modules\eslint\lib\config\config-validator.js:161:30)
    at Object.validate (H:\projects\node\carabiner\node_modules\eslint\lib\config\config-validator.js:239:5)
    at loadFromDisk (H:\projects\node\carabiner\node_modules\eslint\lib\config\config-file.js:507:19)
    at Object.load (H:\projects\node\carabiner\node_modules\eslint\lib\config\config-file.js:550:20)
    at Config.getLocalConfigHierarchy (H:\projects\node\carabiner\node_modules\eslint\lib\config.js:228:44)
    at Config.getConfigHierarchy (H:\projects\node\carabiner\node_modules\eslint\lib\config.js:180:43)
    at Config.getConfigVector (H:\projects\node\carabiner\node_modules\eslint\lib\config.js:287:21)
Process finished with exit code -1

```
When I comment out one-var, it just goes to the next one where I specify additional options in an object. I've looked over the documentation multiple times and can't find any discrepancies which would cause this to be throwing errors. Am I perhaps misunderstanding how configuration works?

**Comment by platinumazure** [comment]
The configuration for `one-var` looks good to me.

Are you quite positive you've configured WebStorm to point at your installed ESLint? Some editor configurations will fall back to a pre-bundled ESLint when they can't find your installed ESLint, and that bundled ESLint could be old (so some rules' newer options wouldn't work).

**Comment by ArcticZeroo** [comment]
Yup, as far as I can tell it's properly configured - http://frozor.io/up/WD8lvgCe.png
The stacktrace also points back to the node_modules folder where I would expect it to be installed, so I'm almost certain it is properly set up

**Comment by ArcticZeroo** [comment]
It is worth noting that I tried to add some console.log statements in the stacktrace locations to help see if I could pinpoint a bug, but nothing additional printed. Could be that webstorm filtered it out, though.

**Comment by platinumazure** [comment]
Agreed, that looks like a correct configuration.

Isn't it intriguing that one-var is the rule that is flagged (and then later rules, once you comment out one-var)? I'm a bit curious as to why earlier rules with options, such as indent, are not also flagged.

**Comment by ArcticZeroo** [comment]
Yeah, I noticed that as well. I'm not really sure what could be causing that! The rule for yoda was also getting flagged, but that mysteriously went away randomly. I'm a bit confused as to what's going on here...

**Comment by platinumazure** [comment]
The only thing I can think of at this point is, maybe some npm issue resulted in an inconsistent set of node_modules somehow so that your package dependencies are not being (consistently) met. Maybe you could remove your node_modules folder and run `npm install` (or equivalent) and hopefully things will magically start working?

**Comment by ArcticZeroo** [comment]
Unfortunately I'm getting the same error after deleting it and re-installing! 

My npm installation is also half broken right now so I wouldn't really doubt it's that to be honest. I can't even install eslint when node_modules already exists, npm is exiting with some error code that has no solutions online that work. I suppose I'll try to fix that first and then will come back to see if this still happens...

**Comment by ArcticZeroo** [comment]
Using yarn to install eslint instead, now one-var is apparently valid, but it's moved onto the next one (require-jsdoc) as being invalid...

**Comment by ArcticZeroo** [comment]
Alright, I've managed to track down the issues with some debug. 

- one-var's consecutive support for initialized and uninitialized were added after 4.18.2, updating ESLint removes the error
- require-jsdoc needs a "require" property with further nested options for some reason. I must have skipped over that part in the docs and put all the properties at the highest object level.

It looks like after updating to the newest ESLint, there are no further issues. This was definitely a configuration problem, but it would be nice to note in the docs that the consecutive option was added later than the rule for other peoples' sanity.
